### ISSUE TITLE
Do notation doesn't work for children classes

### ISSUE THREAD

**Issue by korolvs** [dry-rb/dry-monads#68]
Hi, thank you for a great Gem!
I have such problem:
```ruby
class BaseCommand
  include Dry::Monads::Result::Mixin
  include Dry::Monads::Do.for(:call)
  
  def validate
    Success(:ok)
  end
end

class CreateCommand < BaseCommand
  def call
    yield validate
  end
end
```
Expected behaviour:
```
:ok
```
Actual behaviour:
```
LocalJumpError: no block given (yield)
```

Looks like this is broken because current solution(prepending module that overrides selected methods and makes `yield` work without block) works only for parent class and children have their own class in the first place in the method lookup chain

Do you have making it works for children in your plans?
Thanks!

P.S. If I understand the problem correctly then this solution might work
https://stackoverflow.com/questions/49897647/ruby-prepend-module-how-to-prepend-module-to-the-beginning-of-ancestors-array
I can try it a bit later

**Comment by flash-gordon** [comment]
It doesn't work for subclasses because `.for` is not aware of them. If you want this, use the `inherited` hook in the way described in the link you posted. Another solution is to go with the fully automated version of `Do`, this will do the trick:

```ruby
include Dry::Monads::Do
```

^it will wrap all the methods defined after this line, including subclasses

**Comment by flash-gordon** [comment]
@korolvs oh, you're right and it's rather unexpected. I ditched inheritance completely that's why I haven't come across the issue. I'll fix it tomorrow since it'll require some testing before being sure it's fixed properly

**Comment by jdwolk** [comment]
Just a PSA for people who still get `LocalJumpError: no block given (yield)` despite using `include Dry::Monads::Do`:

Double-check that your base class isn't defining `self.inherited` somewhere. I had code like this:

```
class MyBaseClass
  include Dry::Monads[:result]
  include Dry::Monads::Do

  def.self.inherited(subclass)
      ...
   end
end

class Subclass < MyBaseClass
  def call
    yield foobar
  end
end
```

I was getting this issue until I found out `Dry::Monads::Do::All` is doing some metaprogramming with `self.inherited`: https://github.com/dry-rb/dry-monads/blob/master/lib/dry/monads/do/all.rb#L70

The fix was to call `super` in `self.inherited` on the base class, i.e.:

```
class MyBaseClass
  include Dry::Monads[:result]
  include Dry::Monads::Do

  def.self.inherited(subclass)
      super(subclass)

      ...
   end
end

class Subclass < MyBaseClass
  def call
    yield foobar
  end
end
```

**Comment by lcolombini** [comment]
Hi,
I have the similar error on github actions :LocalJumpError: no block given (yield)
When i run the CI on github actions, the keyword 'yield' by Dry monads not works. WHen I run locally all tests are ok. What is the reason?

**Comment by flash-gordon** [comment]
@lcolombini pls ask questions on our forum (see contributing.md for the link). Please also attach example code that doesn't work, at least essential parts. Otherwise, I can't what's wrong.

**Comment by lcolombini** [comment]
@flash-gordon the sample is a classic dry-monads construct:
`def call
      value = yield get_value_func
  end
`
When I ran CI on GitHub, the error was LocalJumpError: no block given (yield), but I undestand the problem I the library Dry-Monads v1.4 in latest Ubuntu.
I'm developing on Mac OS and I don't have errors. I switched on Ubuntu machine for a trial and I replicated the issue. I solved back to v.1.3.5
